src/waveform/renderers/waveformmark.h

```diff
@@ -1,39 +1,35 @@
 #ifndef WAVEFORMMARK_H
 #define WAVEFORMMARK_H
 
-#include <QString>
+#include <QDomNode>
 #include <QImage>
-#include <QColor>
 
-#include "preferences/usersettings.h"
-#include "skin/skincontext.h"
+#include "util/memory.h"
+
+#include "waveform/renderers/waveformmarkproperties.h"
 
 class ControlProxy;
-class QDomNode;
+class SkinContext;
 class WaveformSignalColors;
 
 class WaveformMark {
   public:
     WaveformMark();
-    ~WaveformMark();
+
     void setup(const QString& group, const QDomNode& node,
                const SkinContext& context,
                const WaveformSignalColors& signalColors);
-    void setKeyAndIndex(const ConfigKey& key, int i);
 
-  private:
-    ControlProxy* m_pPointCos;
+    std::unique_ptr<ControlProxy> m_pPointCos;
 
-    QColor m_color;
-    QColor m_textColor;
-    QString m_text;
-    Qt::Alignment m_align;
-    QString m_pixmapPath;
+    const WaveformMarkProperties& properties() const;
```

Comment:
It would be nice to have a rule for this.

On 21.08.2016 14:19, Ferran Pujol Camins wrote:
>
> In src/waveform/renderers/waveformmark.h 
> <https://github.com/mixxxdj/mixxx/pull/992#discussion_r75594394>:
>
> >
> > -    QColor m_color;
> > -    QColor m_textColor;
> > -    QString m_text;
> > -    Qt::Alignment m_align;
> > -    QString m_pixmapPath;
> > +    const WaveformMarkProperties& properties() const;
>
> I've checked both ours and Google style guide and there's nothing about 
> getter names. Maybe I can add a section to the style guide about this?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/992/files/2f8863b7b8266565a97581fd38cfa6e87ab0ac36#r75594394>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABZ-aqS9L0Dbzlc9_ZeJvvbWWr35aLpeks5qiEJngaJpZM4Jj7vm>.
>